notes.txt

```diff
@@ -0,0 +1,241 @@
+
+I vote for (2) for its simplicity, but (3) could be interesting down the line bc it allows 
+unbondings in less than 21 days
+
+# Records we need to keep
+
+
+
+
+*******************************************************************
+******************************DEPOSITS*****************************
+*******************************************************************
+
+# SECTION 1
+# 0 Update DepositRecord type
+  UPDATE DEPOSIT RECORD
+  message DepositRecord {
+    uint64 id = 1; // epoch_num:chain_id
+    int64 amount = 2; // total amount
+    string denom = 3; // tokens
+    string hostZoneId = 4; // can remove - fetch with denom
+    // the next thing we should do
+    enum RecordType {
+      DEPOSIT = 0;
+      REWARDS = 1;
+      WITHDRAWAL = 2;
+    }
+    enum Status {
+      TRANSFER = 0;
+      STAKE = 1;
+    }
+    Status status = 5; 
+  }
+  - Add next status types
+      - transfer: the balance must be transferred to the host zone
+      - stake: the balance must be staked
+# 1 Aggregate into a single deposit record / successful epoch / chain
+# 2 Update DepositRecord type
+# 3 Update Epoch processing code
+
+>>> EPOCH START
+----------
+
+# users call liquid stake
+def liquidStake(tokens)
+  currentRecord = getRecord(epochId)
+  currentRecord.amount += tokens.amount
+
+>>> EPOCH END
+for record in depositRecords:
+  hz = getHz(record)
+  if record.status == TRANSFER:
+    def ibcTransferAck(ack):
+      if ack == success:
+        record = getRecord(ack)
+        delete(record)
+    # add record deletion logic to ack
+    ibcTransfer.ackFunction = ibcTransferAck
+    # ibc transfer the tokens over
+    ibcTransfer(to=hz.depositAccount(), amount=record.amount)
+
+----------
+>>> NEXT EPOCH STARTS
+
+
+
+
+
+
+
+
+
+*******************************************************************
+******************************REWARDS******************************
+*******************************************************************
+
+# SECTION 2
+# 0 Set withdrawal address of delegation account to withdrawal account in host zone
+    registration ack
+# 1 Update DepositRecord to handle rewards types
+# 2 Update epoch end to process all records for each host zone
+# 3 Add new records for rewards that accrued in the previous epoch
+
+>>> EPOCH START
+----------
+... rewards accrue via calls to liquid stake
+
+>>> EPOCH END
+
+# process all unprocessed records for each host zone
+for record in records where record.type == REWARD:
+  def bankSendAck(ack):
+    record = getRecord(ack.id)
+    record.type = DEPOSIT
+    record.status = STAKE
+  # do upon completion
+  ICABankSend.Ack = bankSendAck
+  ICABankSend(from=withdrawal, to=delegat, amount=record.amount)
+
+# add new records for rewards that accrued in the previous epoch
+for hz in getAllHostZones():
+  def addRewardsRecord(hz):
+    # define a query callback function
+    def queryCB(hz):
+      recordBalance = sumAllRewardsRecords()
+      newBalance = balance - recordBalance
```

Comment:
why are you subtracting these here?

Reply:
Don't we need to exclude the stride commission from the rewardRecord amount? Seems like the pseudocode queries for the full balance of the withdrawal account and adds a record that will sweep the entire balance back to the delegation account (that'd mean stride gets no commission!)